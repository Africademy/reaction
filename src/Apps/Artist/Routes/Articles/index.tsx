import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArticlesRefetchContainer as Articles } from "./ArticlesRefetchContainer"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist as any} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  graphql`
    fragment Articles_artist on Artist {
      ...ArticlesRefetchContainer_artist
    }
  `
)
