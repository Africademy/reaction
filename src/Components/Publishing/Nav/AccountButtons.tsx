import React from "react"
import styled from "styled-components"
import StyledButton from "../../Buttons/Default"
import InvertedButton from "../../Buttons/Inverted"
import { pMedia } from "../../Helpers"
import { Fonts } from "../Fonts"
import { IconHamburger } from "../Icon/IconHamburger"

export const AccountButtons: React.SFC<null> = props => {
  return (
    <AccountButtonsContainer>
      <Link className="mlh-login" href="/log_in">
        <Login>
          Login
        </Login>
      </Link>
      <Link className="mlh-signup" href="/sign_up">
        <Signup>
          Signup
        </Signup>
      </Link>
      {/* TODO: Implement a dropdown nav once design is ready */}
      <MobileNav>
        <IconHamburger color="white"/>
      </MobileNav>
    </AccountButtonsContainer>
  )
}

const Login = InvertedButton.extend`
  ${Fonts.avantgarde("s11")}
  height: 30px;
  width: 80px;
  border-radius: 2px;
`
const Signup = StyledButton.extend`
  ${Fonts.avantgarde("s11")}
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;  
`
const Link = styled.a`
  text-decoration: none;
`
const MobileNav = styled.div`
  margin-right: 5px;
  display: none;
`
const AccountButtonsContainer = styled.div`
  ${pMedia.sm`
    ${Login}, ${Signup} {
      display: none;
    }
    ${MobileNav} {
      display: flex;
    }
  `}
`
