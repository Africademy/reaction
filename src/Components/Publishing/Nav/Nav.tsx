import React from "react"
import styled, { StyledFunction } from "styled-components"
import StyledButton from "../../Buttons/Default"
import InvertedButton from "../../Buttons/Inverted"
import Icon from "../../Icon"
import { Fonts } from "../Fonts"
import { PartnerInline } from "../Partner/PartnerInline"

interface Props {
  article: any
  sponsored?: boolean
  transparent?: boolean
}

interface DivProps {
  transparent: boolean
}

export const Nav: React.SFC<Props> = props => {
  const {
    sponsored,
    transparent
  } = props
  return (
    <NavContainer transparent={transparent}>
      <div>
        {sponsored ?
          <PartnerInline
            url="/"
            logo="https://artsy-media-uploads.s3.amazonaws.com/kq-CcNCHEgAuPadHtOveeg%2FPlanetArt_Black.png"
            color="white"
          />
          :
          <a href='/'>
            <Icon
              name="logo"
              color="white"
              fontSize="32px"
            />
          </a>
        }
      </div>
      <Title>Artsy Editorial</Title>
      <div>
        <Login>Login</Login>
        <Signup>Signup</Signup>
      </div>
    </NavContainer>
  )
}

Nav.defaultProps = {
  sponsored: false,
  transparent: false
}

const Div: StyledFunction<DivProps & React.HTMLProps<HTMLDivElement>> = styled.div
const NavContainer = Div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 46px;
  width: 100%;
  background-color: ${props => props.transparent ? "transparent" : "black"};
  color: white;
`
const Title = styled.div`
  ${Fonts.garamond("s17")}
`
const Login = InvertedButton.extend`
  ${Fonts.avantgarde("s11")}
  height: 30px;
  width: 80px;
  border-radius: 2px;
`
const Signup = StyledButton.extend`
  ${Fonts.avantgarde("s11")}
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;  
`
