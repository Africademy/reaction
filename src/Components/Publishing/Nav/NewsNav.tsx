import moment from "moment"
import React from "react"
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { Fonts } from "../Fonts"

interface Props {
  date: string
}

export const NewsNav: React.SFC<Props> = props => {
  const { date } = props
  const today = new Date()
  const hasYear = moment(date).format("YYYY") !== moment(today).format("YYYY")
  const isToday =
    moment(date).format("MMM D, YYYY") === moment(today).format("MMM D, YYYY")
  const format = hasYear ? "MMM D, YYYY" : "MMM D"

  return (
    <NewsNavContainer>
      <MaxWidthContainer>
        <NavText>{isToday ? "Today" : moment(date).format(format)}</NavText>
        <Title>The News</Title>
      </MaxWidthContainer>
    </NewsNavContainer>
  )
}

const NavText = styled.div`
  ${Fonts.unica("s25", "medium")};
  ${pMedia.sm`
    ${Fonts.unica("s16", "medium")}
  `};
`

const Title = NavText.extend`
  position: absolute;
  left: 30px;
  ${pMedia.sm`
    left: 20px;
  `};
`

const MaxWidthContainer = styled.div`
  position: relative;
  max-width: 780px;
  margin: auto;
  display: flex;
  justify-content: center;
`

const NewsNavContainer = styled.div`
  position: fixed;
  width: 100%;
  box-shadow: 0 0 8px 0 rgba(0, 0, 0, 0.2);
  padding: 10px 0;
  background: white;
  z-index: 1;
`
