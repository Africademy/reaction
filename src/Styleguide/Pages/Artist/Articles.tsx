import React from "react"
import {
  graphql,
  QueryRenderer,
  createRefetchContainer,
  RelayRefetchProp,
} from "react-relay"
import { ContextProps, ContextConsumer } from "Components/Artsy"
import styled from "styled-components"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Separator } from "Styleguide/Elements/Separator"
import { ArticleItem } from "./ArticleItem"
import { Pagination } from "Styleguide/Components/Pagination"

import { Articles_artist } from "__generated__/Articles_artist.graphql"

interface ArticlesProps {
  relay: RelayRefetchProp
  artist: Articles_artist
}

const PAGE_SIZE = 10

// Renders list of articles, we'll use this in our refetch container.
class ArticlesContainer extends React.Component<ArticlesProps> {
  loadPrev = () => {
    const {
      startCursor,
      hasPreviousPage,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasPreviousPage) {
      this.loadBefore(startCursor)
    }
  }

  loadNext = () => {
    const {
      hasNextPage,
      endCursor,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadBefore(cursor) {
    this.props.relay.refetch(
      {
        first: null,
        before: cursor,
        artistID: this.props.artist.id,
        after: null,
        last: PAGE_SIZE,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  loadAfter = cursor => {
    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  renderPagination() {
    return (
      <div>
        <Pagination
          {...this.props.artist.articlesConnection.pageCursors}
          onClick={this.loadAfter}
          onNext={this.loadNext}
          onPrev={this.loadPrev}
        />
      </div>
    )
  }

  render() {
    return (
      <div>
        <Row>
          <Col>
            <ArticleList>
              {this.props.artist.articlesConnection.edges.map(({ node }) => {
                return (
                  <ArticleItem
                    title={node.thumbnail_title}
                    imageUrl={node.thumbnail_image.resized.url}
                    date={node.published_at}
                    author={node.author.name}
                    href={node.href}
                  />
                )
              })}
            </ArticleList>
          </Col>
        </Row>

        <Box my={2}>
          <Separator />
        </Box>

        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              {this.renderPagination()}
            </Flex>
          </Col>
        </Row>
      </div>
    )
  }
}

// This is the actual Refetch Container we want to use.
const ArticlesRefetchContainer = createRefetchContainer(
  ArticlesContainer,
  {
    artist: graphql`
      fragment Articles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int" }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          pageCursors {
            around {
              cursor
              page
              isCurrent
            }
            first {
              cursor
              page
              isCurrent
            }
            last {
              cursor
              page
              isCurrent
            }
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...Articles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)

interface Props extends ContextProps {
  artistID: string
}

class ArticlesQueryRenderer extends React.Component<Props> {
  render() {
    const { artistID, relayEnvironment } = this.props
    return (
      <QueryRenderer
        environment={relayEnvironment}
        query={graphql`
          query ArticlesIndexQuery($artistID: String!, $first: Int!) {
            artist(id: $artistID) {
              ...Articles_artist @arguments(first: $first)
            }
          }
        `}
        variables={{ artistID, first: PAGE_SIZE }}
        render={({ props }) => {
          if (props) {
            return <ArticlesRefetchContainer artist={props.artist} />
          } else {
            return null
          }
        }}
      />
    )
  }
}

export const ArticlesContent = ContextConsumer(ArticlesQueryRenderer)

const ArticleList = styled.div``
