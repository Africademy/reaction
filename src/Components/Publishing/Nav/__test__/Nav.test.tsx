import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { FeatureArticle } from "../../Fixtures/Articles"
import { Nav } from "../Nav"

describe("Nav", () => {
  it("renders a Nav", () => {
    const nav = renderer.create(<Nav article={FeatureArticle} />)
    expect(nav).toMatchSnapshot()
  })
  it("renders a transparent nav", () => {
    const nav = renderer.create(<Nav article={FeatureArticle} transparent/>)
    expect(nav).toMatchSnapshot()
  })
  it("renders a sponsored nav", () => {
    const nav = renderer.create(<Nav article={FeatureArticle} sponsored/>)
    expect(nav).toMatchSnapshot()
  })
})
