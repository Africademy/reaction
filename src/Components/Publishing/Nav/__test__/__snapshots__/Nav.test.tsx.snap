// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders a Nav 1`] = `
.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 32px;
  vertical-align: middle;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c1 .IconPlus {
  margin: 0 20px;
}

.c9 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c9:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c8 {
  background: black;
  color: white;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  border-radius: 2px;
}

.c8:hover:not(:disabled) {
  background: #6e1fff;
}

.c11 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
}

.c11:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c6 {
  text-decoration: none;
}

.c13 {
  margin-right: 5px;
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 46px;
  width: 100%;
  background-color: black;
  color: white;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .IconPlus {
    margin: 0 15px;
  }
}

@media (max-width:720px) {
  .c5 .c7,
  .c5 .c10 {
    display: none;
  }

  .c5 .c12 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="c0"
>
  <div
    className="PartnerInline c1"
  >
    <a
      href="/"
    >
      <div
        className="c2 c3"
        style={undefined}
      >
        î˜ˆ
      </div>
    </a>
  </div>
  <div
    className="c4"
  >
    Artsy Editorial
  </div>
  <div
    className="c5"
  >
    <a
      className="mlh-login c6"
      href="/log_in"
    >
      <button
        className="c7 c8 c9"
      >
        <span>
          Login
        </span>
      </button>
    </a>
    <a
      className="mlh-signup c6"
      href="/sign_up"
    >
      <button
        className="c10 c11"
      >
        <span>
          Signup
        </span>
      </button>
    </a>
    <div
      className="c12 c13"
    >
      <svg
        enable-background="new 0 0 32 32"
        fill="white"
        version="1.1"
        viewBox="0 0 32 32"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="2"
          width="20"
          x="6"
          y="22"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="15"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="8"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Nav renders a sponsored nav 1`] = `
.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c4 {
  width: 15px;
  height: 15px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 32px;
  vertical-align: middle;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c1 .IconPlus {
  margin: 0 20px;
}

.c10 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c10:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c9 {
  background: black;
  color: white;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  border-radius: 2px;
}

.c9:hover:not(:disabled) {
  background: #6e1fff;
}

.c12 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
}

.c12:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c7 {
  text-decoration: none;
}

.c14 {
  margin-right: 5px;
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 46px;
  width: 100%;
  background-color: black;
  color: white;
}

.c5 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
}

@media (max-width:900px) {
  .c4 {
    width: 13px;
    height: 13px;
  }
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .IconPlus {
    margin: 0 15px;
  }
}

@media (max-width:720px) {
  .c6 .c8,
  .c6 .c11 {
    display: none;
  }

  .c6 .c13 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="c0"
>
  <div
    className="PartnerInline c1"
  >
    <a
      href="/"
    >
      <div
        className="c2 c3"
        style={undefined}
      >
        î˜ˆ
      </div>
    </a>
    <svg
      className="IconPlus c4"
      style={
        Object {
          "margin": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 15 15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="2"
      >
        <g
          fill="white"
          transform="translate(-101.000000, -18.000000)"
        >
          <g
            transform="translate(101.000000, 18.000000)"
          >
            <rect
              height="1"
              width="15"
              x="0"
              y="7"
            />
            <rect
              height="1"
              transform="translate(7.500000, 7.500000) rotate(-90.000000) translate(-7.500000, -7.500000) "
              width="15"
              x="0"
              y="7"
            />
          </g>
        </g>
      </g>
    </svg>
    <a
      href={undefined}
      onClick={[Function]}
      target="_blank"
    >
      <img
        src="https://artsy-media-uploads.s3.amazonaws.com/F4RVgsSXv3q9Nrt59P8gbA%2Ffullscreen.png"
      />
    </a>
  </div>
  <div
    className="c5"
  >
    Artsy Editorial
  </div>
  <div
    className="c6"
  >
    <a
      className="mlh-login c7"
      href="/log_in"
    >
      <button
        className="c8 c9 c10"
      >
        <span>
          Login
        </span>
      </button>
    </a>
    <a
      className="mlh-signup c7"
      href="/sign_up"
    >
      <button
        className="c11 c12"
      >
        <span>
          Signup
        </span>
      </button>
    </a>
    <div
      className="c13 c14"
    >
      <svg
        enable-background="new 0 0 32 32"
        fill="white"
        version="1.1"
        viewBox="0 0 32 32"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="2"
          width="20"
          x="6"
          y="22"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="15"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="8"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Nav renders a transparent nav 1`] = `
.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c1 img {
  max-height: 32px;
  vertical-align: middle;
}

.c1 .c2 {
  margin: 0;
  vertical-align: middle;
}

.c1 .IconPlus {
  margin: 0 20px;
}

.c9 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c9:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c8 {
  background: black;
  color: white;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  border-radius: 2px;
}

.c8:hover:not(:disabled) {
  background: #6e1fff;
}

.c11 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
}

.c11:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c6 {
  text-decoration: none;
}

.c13 {
  margin-right: 5px;
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 46px;
  width: 100%;
  background-color: transparent;
  color: white;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
}

@media (max-width:720px) {
  .c1 img {
    max-height: 24px;
  }

  .c1 .c2 {
    font-size: 24px;
  }

  .c1 .IconPlus {
    margin: 0 15px;
  }
}

@media (max-width:720px) {
  .c5 .c7,
  .c5 .c10 {
    display: none;
  }

  .c5 .c12 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="c0"
>
  <div
    className="PartnerInline c1"
  >
    <a
      href="/"
    >
      <div
        className="c2 c3"
        style={undefined}
      >
        î˜ˆ
      </div>
    </a>
  </div>
  <div
    className="c4"
  >
    Artsy Editorial
  </div>
  <div
    className="c5"
  >
    <a
      className="mlh-login c6"
      href="/log_in"
    >
      <button
        className="c7 c8 c9"
      >
        <span>
          Login
        </span>
      </button>
    </a>
    <a
      className="mlh-signup c6"
      href="/sign_up"
    >
      <button
        className="c10 c11"
      >
        <span>
          Signup
        </span>
      </button>
    </a>
    <div
      className="c12 c13"
    >
      <svg
        enable-background="new 0 0 32 32"
        fill="white"
        version="1.1"
        viewBox="0 0 32 32"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="2"
          width="20"
          x="6"
          y="22"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="15"
        />
        <rect
          height="2"
          width="20"
          x="6"
          y="8"
        />
      </svg>
    </div>
  </div>
</div>
`;
